src/app/shared/comment-editor/comment-editor.component.ts

```diff
@@ -18,14 +18,20 @@ export class CommentEditorComponent implements OnInit {
 
   @Input() commentField: AbstractControl;
   @Input() commentForm: FormGroup;
+  @Input() id?: string;
```

Comment:
The other phases use the `comment-editor` as a static form so like `AbstractControl` references the comment editor by the `description` identifier. But for the dynamic case we can't do that because if all the editors have the identifier of `description`, changing a value in the `AbstractControl` will change the values for all the editors. Making it compulsory might cause the other phases to break.

Also if i recall, removing the `?` doesn't cause any compiler errors nor does the IDE prompt if the dev don't pass in an `id`.

Reply:
I was thinking that making `id` a compulsory attribute will make the `comment-editor` more generic. Not sure how it would break other phases tho, since we just need to add an additional `id` attribute to all the usage of `comment-editor`. 

However, as you said, Angular doesn't have the concept of compulsory `@Input`. Maybe we could do some sort of validation during initialization and documentation to specify what are the required inputs.

Reply:
> However, as you said, Angular doesn't have the concept of compulsory `@Input`. Maybe we could do some sort of validation during initialization and documentation to specify what are the required inputs.

@JunWei96 

Ahh I see, I can just input the `[id]='description'` into the comment editor for previous phases to get it to work. And perhaps do a ...
```typescript
if (this.id === undefined) {
  throw new Error("Comment Editor's FormName (ID) has not been specified");
}
```
This way the app will throw and error on runtime. Would this be okay?

Reply:
Sounds good :)